pydata/xarray
# join together duplicate entries in the text `repr`

`Indexes` contains one entry per coordinate, even if there are indexes that index multiple coordinates. This deduplicates the entries, but the format is not quite clear.

- [x] follow-up to #6795

The formatting options we were able to come up with:
1. separate with just newlines (see e29aeb9085bc677e16dabd4a2b94cf63d06c155e):
```
Indexes:
    one    CustomIndex
    two
    three  PandasIndex
```
2. mark unique indexes with a prefix to make it look like a list (see 9b90f8bceda6f012c863927865cc638e0ff3fb88):
```
Indexes:
  - one    CustomIndex
    two
  - three  PandasIndex
```
3. use unicode box components (in front of the coordinate names) (see 2cec070ebf4f1958d4ffef98d7649eda21ac09a3):
```
Indexes:
  ┌ one    CustomIndex
  │ two
  └ three
    four  PandasIndex
```
4. use unicode box components (after the coordinate names) (see 492ab47ccce8c43d264d5c759841060d33cafe4d):
```
Indexes:
    one   ┐ CustomIndex
    two   │
    three ┘
    four    PandasIndex
```

For the unicode box components, we can choose between the light and heavy variants.

@benbovy and I think the unicode variants (especially variant 3) are the easiest to understand, but we would need to decide whether we care about terminals that don't support unicode.

Edit: in the meeting we decided that support for the subsection of unicode should be common enough that we can use it. I'll clean this PR up to implement option 3, then.
